### PR TITLE
Remove glitch when displaying platform views

### PR DESCRIPTION
**Pull request by blasten**
When `startRenderingToSurface` is called, any current active surface is destroyed.

This is undesired when platform views are displayed because the Engine will continue
to render to this surface once the platform views aren't rendered in the frame.

If the active surface is destroyed, there's a "glitch" caused by the time it takes to create
the new surface.

This issue was introduced after https://github.com/flutter/engine/pull/28894

Fixes https://github.com/flutter/flutter/issues/95343
Fixes https://github.com/flutter/flutter/issues/96177

### CODE REVIEW

**Comment by zanderso** [comment]
The failing unit tests appear to be related.

**Comment by blasten** [comment]
The test should be passing now. ptal

File: shell/platform/android/io/flutter/embedding/android/FlutterSurfaceView.java [hunk 1]
```diff
@@ -253,8 +256,17 @@ private void connectSurfaceToRenderer() {
...
           "connectSurfaceToRenderer() should only be called when flutterRenderer and getHolder() are non-null.");
     }
-
-    flutterRenderer.startRenderingToSurface(getHolder().getSurface());
+    // When connecting the surface to the renderer, it's possible that the surface is currently
+    // paused. For instance, when a platform view is displayed, the current FlutterSurfaceView
+    // is paused, and rendering continues in a FlutterImageView buffer while the platform view
+    // is displayed.
+    //
+    // startRenderingToSurface may stop rendering to this surface if it believes it already started
```
Comment (GaryQian) [inline shell/platform/android/io/flutter/embedding/android/FlutterSurfaceView.java:264]: Nit, the wording here is a bit confusing? Not sure if can be better said.

**Review by GaryQian — APPROVED** [review]
LGTM!

File: shell/platform/android/io/flutter/embedding/android/FlutterSurfaceView.java (hunk 1 above)
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/android/FlutterSurfaceView.java]: agreed. Done

**Pull request merged**